hooks/pre_commit/14-mongo-cache-size.rb

```diff
@@ -0,0 +1,13 @@
+# Setting Default Mongo WiredTiger Engine cache size to be 20% of total memory
+# See https://access.redhat.com/solutions/4505561
+# See https://docs.mongodb.com/manual/core/wiredtiger/#memory-use
+MONGO_CACHE_SIZE_FACT = 'mongo_cache_size'.freeze
+
+if (current_mongo_cache_size = get_custom_fact(MONGO_CACHE_SIZE_FACT))
+  required_mongo_cache_size = (facts[:memory][:system][:total_bytes] * 0.2 / 1024 / 1024 / 1024).round(2)
+  if current_mongo_cache_size != required_mongo_cache_size
+    store_custom_fact(MONGO_CACHE_SIZE_FACT, required_mongo_cache_size)
+    # Store the app config to disk
+    kafo.config.configure_application
+  end
+end
```

Comment:
Could you hide this behind `pulp_enabled?`? That way we don't run this on Debian and when Pulp 2 has been turned off.